src/controllers/room.controller.js

```diff
@@ -0,0 +1,82 @@
+import { Room } from '../models/Room.models.js';
+
+export const getRooms = async (req, res) => {
+  res.send(await Room.findAll());
+};
+
+export const getOneRoom = async (req, res) => {
+  const { id } = req.params;
+  const room = await Room.findByPk(id);
+
+  if (!room) {
+    res.status(404).send({ message: 'cant get a room' });
+
+    return;
+  }
+  res.status(200);
+  res.send(room);
+};
+
+export const createRoom = async (req, res) => {
+  const { roomName } = req.body;
+
+  if (!roomName) {
+    res.status(400).send({ message: 'room name is required' });
+  }
+
+  const room = {
+    roomName,
+  };
+
+  const newRoom = await Room.create(room);
+
+  res.status(201);
+  res.send(newRoom);
+};
+
+const deleteR = async (id) => {
+  await Room.destroy({
+    where: {
+      id,
+    },
+  });
+};
+
+export const deleteRoom = async (req, res) => {
+  const { id } = req.params;
+
+  if (!(await Room.findByPk(id))) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  await deleteR(id);
+  res.sendStatus(204);
+};
+
+const patchRoom = async ({ roomName, id }) => {
+  await Room.update({ roomName }, { where: { id } });
+
+  return Room.findByPk(id);
+};
+
+export const renameRoom = async (req, res) => {
+  const { id } = req.params;
+  const { roomName } = req.body;
+
+  if (!(await Room.findByPk(id))) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const updatedRomm = await patchRoom({ roomName, id });
+
+  if (!updatedRomm) {
+    res.sendStatus(404).send({ message: 'cant update a room' });
```

Comment:
In your `renameRoom` function, you are trying to set the status code to 404 and send a response at the same time. However, `res.sendStatus` ends the response and `res.send` cannot be called after that. If you want to send a custom message with a 404 status code, you should use `res.status(404).send({ message: 'cant update a room' });`